packages/site/cms-site/src/sitePreview/SitePreviewUtils.ts

```diff
@@ -1,4 +1,4 @@
-//TODO add import "server-only"; once this file gets correctly tree-shaked out of the client bundle
+import "server-only";
```

Comment:
This would be a breaking change since the code is currently being added to the client bundle, whether one migrates to App Router or not. Should we move this post v7?

Reply:
Is the situation improved for v8?

Reply:
No, not really. But v7 is very close to a release. @thomasdax98 what do you think? When considering page speed, it would be good to add this.

Reply:
We can still add it to v7.0, as we haven't released it yet (if we merge it today)

Also this sounds to me like the change is rather theoretically breaking than actually breaking, right? These functions were added in v7 and are meant to be used on the server. So it won't affect any projects.

Reply:
I believe you'll still need to make these two changes

* Update to Next 14.2.0 or later
* Enable [optimizePackageImports](https://nextjs.org/docs/app/api-reference/next-config-js/optimizePackageImports)

Reply:
We decided to ship this in v7.